readers-common/src/main/java/uk/gov/gchq/palisade/reader/common/SerialisedDataReader.java

```diff
@@ -33,14 +33,13 @@
 import static java.util.Objects.requireNonNull;
 
 /**
- * This class is an abstract implementation of the {@link DataReader} which uses
- * serialisers to serialise the data into the for that the rules need to be able
- * to apply those rules and then de-serialise to how the client is expecting the
- * data to be returned.
+ * This class is an abstract implementation of the {@link DataReader} which
+ * serialises the data into the format needed for applying the rules
+ * and then deserialises the data to the format the client is expecting.
```

Comment:
I think this is meant to be the other way round. We deserialise, from avro for example, so that we can apply rules to mask/redact the information and then serialise back when passing to the client.
```suggestion
 * This class is an abstract implementation of the {@link DataReader} which
 * deserialises the data into the format needed for applying the rules
 * and then serialises the data to the format the client is expecting.
```
It might be worth checking this though as I might be wrong.

Reply:
Yep, you got it right. Serialisation will convert a Java into a stream of bytes.  Deserialisation will convert a stream of bytes into an Object.